cmd/main.go

```diff
@@ -168,46 +191,46 @@ func (c Configuration) genericConsumer(fs afero.Fs) []GenericFile {
 		if PathFull == "" {
 			PathFull = fullPath
 		}
-		logger.Debug().Msgf("generic file to watch: %v", PathFull)
+		logger.Debug().Msgf("file to watch: %v", PathFull)
 		PathFull, fi := c.resolvePath(PathFull)
 		if PathFull == "" {
 			continue // could not resolve the file. skip for now.
 		}
-		if c.checkIgnored(PathFull, fs) {
+		if c.checkIgnored(PathFull, fs, consumerType) {
 			continue // skip ignored file
 		}
 		switch mode := fi.Mode(); {
 		case mode.IsDir():
-			logger.Debug().Msg("Generic is dir")
+			logger.Debug().Msg("Path is a dir")
 			err := filepath.Walk(PathFull, func(path string, info os.FileInfo, err error) error {
-				if c.checkIgnored(path, fs) {
+				if c.checkIgnored(path, fs, consumerType) {
```

Comment:
Let's try to get rid of usage of `consumerType` in that `checkIgnored`

Reply:
Removed checkIgnored and added logic to exclude files that are already mapped to consumers from being mapped again.